File: backend/internal/dto/supervision.go
package dto

// DeptRate 是分教研室覆盖率。
type DeptRate struct {
	Department string  `json:"department"`
	Rate       float64 `json:"rate"`
}

// CoverageDTO 是 GET /supervision/coverage 的响应。
type CoverageDTO struct {
	TotalCourses      int        `json:"totalCourses"`
	SupervisedCourses int        `json:"supervisedCourses"`
	Rate              float64    `json:"rate"`
	ByDepartment      []DeptRate `json:"byDepartment"`
}

// PlanItem 是听评课安排行。
type PlanItem struct {
	ID             uint64 `json:"id"`
	CourseID       uint64 `json:"courseId"`
	CourseName     string `json:"courseName"`
	TeacherName    string `json:"teacherName"`
	SupervisorName string `json:"supervisorName"`
	PlannedDate    string `json:"plannedDate"`
	Status         string `json:"status"`
}

// PlanListQuery 是 GET /supervision/plans 的查询参数。
type PlanListQuery struct {
	Status   string `form:"status" binding:"omitempty,oneof=planned completed"`
	DateFrom string `form:"dateFrom"`
	DateTo   string `form:"dateTo"`
	Page     int    `form:"page"`
	PageSize int    `form:"pageSize"`
}

File: frontend/src/types/supervision.ts
export type SupervisionStatus = 'planned' | 'completed'

export interface SupervisionPlan {
  id: number
  courseId: number
  courseName: string
  teacherName: string
  supervisorName: string
  plannedDate: string
  status: SupervisionStatus
}

export interface DepartmentCoverage {
  department: string
  rate: number
}

export interface CoverageStat {
  totalCourses: number
  supervisedCourses: number
  rate: number
  byDepartment: DepartmentCoverage[]
}

export interface SupervisionQuery {
  status?: SupervisionStatus | ''
  /** 计划日期起（在 api 层映射为后端 dateFrom） */
  date?: string
  page?: number
  pageSize?: number
}

export interface DashboardStat {
  label: string
  value: number | string
  unit?: string
  icon?: string
  tone?: 'primary' | 'success' | 'warning' | 'info' | 'supervisor'
}

/**
 * 工作台视图模型。
 * 后端 GET /dashboard 按角色返回三种 DTO，由 api/dashboard.ts 收敛成同一形状，
 * 页面组件因此只需一套模板 + 角色分支。
 */
export interface DashboardData {
  stats: DashboardStat[]
  courses?: import('./course').Course[]
  plans?: SupervisionPlan[]
  coverage?: CoverageStat | null
  recentResources?: import('./resource').Resource[]
}
